File: ProjektyStudentow/2019_jakub_krzeminski/Projekt/Product.php
<?php
/**
 * Created by PhpStorm.
 * User: krzemson
 * Date: 25.01.2019
 * Time: 02:20
 */

namespace App;


class Product
{
    private $name;
    private $basicPrice;
    private $strategyPrice;

    public function __construct($name = "", $basicPrice = null, Price $strategy = null) {
        $this->name=$name;
        $this->basicPrice=$basicPrice;
        $this->strategyPrice = new $strategy();
    }
    public function getPrice()
    {
        return $this->strategyPrice->count($this->basicPrice);
    }
    public function getName()
    {
        return $this->name;
    }
    public function setStrategy($strategy)
    {
        $this->strategyPrice = new $strategy();
    }

    public function getAll()
    {
        echo "Lista produktów\n";
    }

    public function get($id)
    {
        echo "Produkt o ID ".$id."\n";
    }
}

File: 02Fabryka/Students/2019/LempaszekAgnieszka/Factory/src/products/Product.java
package products;

public abstract class Product {
	
	protected Size size;
	protected Taste taste;
	protected int price;
	protected String name;

	public abstract void make();
	public abstract void taste();
	
	public int getPrice() {
		return price;
	}
	
	public Size getSize() {
		return size;
	}

	public String toString() {
		return "Size: " + size + ", Price: " + price + ", Taste: " +  taste;
	}
	
}
